src/BKTClient.ts

```diff
@@ -1,22 +1,61 @@
-import { BKTEvaluation } from './BKTEvaluation'
+import {
+  BKTEvaluation,
+  BKTEvaluationDetail as BKTEvaluationDetails,
+} from './BKTEvaluation'
 import { BKTUser } from './BKTUser'
 import { Component } from './internal/di/Component'
 import { clearInstance, getInstance, setInstance } from './internal/instance'
 import { ApiId } from './internal/model/MetricsEventData'
 import { TaskScheduler } from './internal/scheduler/TaskScheduler'
 import { toBKTUser } from './internal/UserHolder'
+import { BKTValue } from './JsonTypes'
 
 export interface BKTClient {
   stringVariation: (featureId: string, defaultValue: string) => string
   numberVariation: (featureId: string, defaultValue: number) => number
   booleanVariation: (featureId: string, defaultValue: boolean) => boolean
-  jsonVariation: <T>(featureId: string, defaultValue: T) => T
+  /**
+   * @deprecated use objectVariation(featureId: string, defaultValue: BKTJsonValue) instead.
+   */
+  jsonVariation: (featureId: string, defaultValue: BKTValue) => BKTValue
+  objectVariation: (featureId: string, defaultValue: BKTValue) => BKTValue
+
   track: (goalId: string, value: number) => void
   currentUser: () => BKTUser
   updateUserAttributes: (attributes: Record<string, string>) => void
   fetchEvaluations: (timeoutMillis?: number) => Promise<void>
   flush: () => Promise<void>
+  /**
+   * @deprecated use stringVariationDetails(featureId: string, defaultValue: string) instead.
+   */
   evaluationDetails: (featureId: string) => BKTEvaluation | null
+
+  stringVariationDetails: (
+    featureId: string,
+    defaultValue: string,
+  ) => BKTEvaluationDetails<string>
+  numberVariationDetails: (
+    featureId: string,
+    defaultValue: number,
+  ) => BKTEvaluationDetails<number>
+  booleanVariationDetails: (
+    featureId: string,
+    defaultValue: boolean,
+  ) => BKTEvaluationDetails<boolean>
+  /**
+   * Retrieves the evaluation details for a given feature based on its ID.
+   *
+   * @param featureId - The unique identifier for the feature.
+   * @param defaultValue - The default value to return if no result is found. This value should be of type `BKTJsonValue`.
+   *
+   * @returns An object of type `BKTEvaluationDetail<BKTJsonValue>` containing the evaluation details.
+   *
+   * Note: The returned value will be either a BKTJsonObject or a BKTJsonArray. If no result is found, it will return the provided `defaultValue`, which can be of any type within `BKTJsonValue`.
```

Comment:
```suggestion
   * Note: The returned value will be a `BKTValue`. If the flag is not found, it will return the provided `defaultValue`, which can be of any type within `BKTValue`.
```